torchvision/models/detection/roi_heads.py

```diff
@@ -591,30 +667,44 @@ def forward(self, features, proposals, image_shapes, targets=None):
 
             losses.update(loss_mask)
 
-        if self.has_keypoint:
+        # keep none checks in if conditional so torchscript will conditionally
```

Comment:
This is another sort of weird result that falls out of not having first class modules.

When we see an if statement that is composed of statically-true `is None` checks, we only compile one branch of the if statement. In this case, the attributes `has_mask` are either a value or None at the point of compilation, instead of an optional, so we just compile one branch of the if statement.

Reply:
So, if we have `FasterRCNN`, for which all `mask` fields are empty, shouldn't we need this same behavior?

The reason why I'm saying this is because the Mask R-CNN and the Keypoint R-CNN branch are basically symmetric, so I'd expect the way to handle the code to be the same